scripts/linters/general_purpose_linter.py

```diff
@@ -744,6 +744,40 @@ def _check_bad_patterns(self):
                 python_utils.PRINT(summary_message)
         return summary_messages
 
+    def _check_newline_at_eof(self):
+        """This function is used to detect newline at the end of file."""
+        if self.verbose_mode_enabled:
+            python_utils.PRINT(
+                'Starting newline at eof check\n'
+                '----------------------------------------')
+        summary_messages = []
+        files_to_lint = self.all_filepaths
+        failed = False
+
+        with linter_utils.redirect_stdout(sys.stdout):
+            for filepath in files_to_lint:
+                file_content = FILE_CACHE.readlines(filepath)
+                if not re.search(r'[^\n]\n', file_content[-1]):
```

Comment:
it will allow a line with `\t\n`, right? [Not sure whether we want this check to be that much strict!]

Reply:
we have another check to catch `\t` so I think we do not need this.